parquet-hadoop/src/main/java/org/apache/parquet/filter2/compat/RowGroupFilter.java

```diff
@@ -38,21 +39,47 @@
  * no filtering will be performed.
  */
 public class RowGroupFilter implements Visitor<List<BlockMetaData>> {
+  private static final int STATISTICS_FIXED_VERSION = 161; // 1.6.1
   private final List<BlockMetaData> blocks;
   private final MessageType schema;
+  private final Boolean skipStatisticsChecks;
 
-  public static List<BlockMetaData> filterRowGroups(Filter filter, List<BlockMetaData> blocks, MessageType schema) {
+  public static List<BlockMetaData> filterRowGroups(Filter filter, ParquetMetadata footer,
```

Comment:
@isnotinvain, is this a public API? It looks like it to me.

If this is a public API, then we need to keep a version of this method that takes a `List<BlockMetaData>` for compatibility. You can deprecate it though.